File: КОМПОНЕНТЫ/OPartner/OPartner/Examples/CBuildr5/exo2xl1.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "ExO2XL1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "OpDbOfc"
#pragma link "OpDbOlk"
#pragma link "OpExcel"
#pragma link "OpModels"
#pragma link "OpOlk2k"
#pragma link "OpOutlk"
#pragma link "OpShared"
#pragma link "OpXL2K"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
    : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::btnExportClick(TObject *Sender)
{
  TCursor oCursor = Screen->Cursor;
  Screen->Cursor = crHourGlass;

  // check all connections
  if (!OpOutlook1->Connected)
    OpOutlook1->Connected = true;
  if (!OpExcel1->Connected)
    OpExcel1->Connected = true;
  if (!OpContactsDataSet1->Active)
    OpContactsDataSet1->Active = true;

  TOpExcelWorkbook* oWorkBook = OpExcel1->Workbooks->Add(); // create a workbook
  TOpExcelWorksheet* oSheet = oWorkBook->Worksheets->Add(); // add a worksheet
  TOpExcelRange* oRange = oSheet->Ranges->Add();            // create range for output
  oRange->Address = "A1";                                   // locate range
  oRange->OfficeModel = OpDataSetModel1;                    // assign the model to the range
  oRange->Populate();                                       // fill the range from the model
  OpExcel1->Visible = true;                                 // show it
  OpExcel1->Interactive = true;                             // Let the user access it
  Screen->Cursor = oCursor;
}
//---------------------------------------------------------------------------

File: КОМПОНЕНТЫ/OPartner/Examples/Delphi/exo2xl1.pas
unit ExO2XL1;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  StdCtrls, OpXL2K, OpExcel, OpModels, OpDbOfc, Db,
  OpDbOlk, OpShared, OpOlk2K, OpOutlk, Buttons;

type
  TForm1 = class(TForm)
    btnExport: TButton;
    OpExcel1: TOpExcel;
    OpOutlook1: TOpOutlook;
    OpDataSetModel1: TOpDataSetModel;
    OpContactsDataSet1: TOpContactsDataSet;
    procedure btnExportClick(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

procedure TForm1.btnExportClick(Sender: TObject);
var
  oCursor: TCursor;
  oWorkBook: TOpExcelWorkbook;
  oSheet: TOpExcelWorksheet;
  oRange: TOpExcelRange;
begin
  oCursor := Screen.Cursor;
  Screen.Cursor := crHourglass;
  try
    // check all connections
    if not OpOutlook1.Connected then
      OpOutlook1.Connected := True;
    if not OpExcel1.Connected then
      OpExcel1.Connected := True;
    if not OpContactsDataSet1.Active then
      OpContactsDataSet1.Active := True;

    with OpExcel1 do
    begin
      oWorkBook := Workbooks.Add;            // create a workbook
      oSheet := oWorkBook.Worksheets.Add;    // add a worksheet
      oRange := oSheet.Ranges.Add;           // create range for output
      oRange.Address := 'A1';                // locate range
      oRange.OfficeModel := OpDatasetModel1; // assign the model to the range
      oRange.Populate;                       // fill the range from the model
      Visible := True;                       // show it
      Interactive := True                    // Let the user access it
    end;
  finally
    Screen.Cursor := oCursor;
  end;
end;

end.
